### PR TITLE
Ch api question 162813038

### PR DESCRIPTION
**Pull request by Barasa2018**

### CODE REVIEW

**Comment by Barasa2018** [comment]
**What does this PR do ?**
It is an api endpoint to fetch a single question

**What are the Description for the tasks to be completed ?**
Ensure the api endpoint to fetch a single question is working

**How should this be manually tested ?**

-Clone this repository in your local machine
-Cd into the repository
-Create a virtual environment env by using the command python3 -m venv env
-Activate the virtual environment . env/bin/activate
-install flask in your virtual environment pip install Flask
-make directory app/api/v1/questions
-Run flask from your terminal
-Copy the url http://127.0.0.1:5000/app/api/v1/question/1 and paste it in postman
-select GET in the left drop down
-Hit Send

**What are the pivotal stories associated with this PR ?**
#162813038

image

**Comment by khwilo** [comment]
Change the title of your PR to conform to the Andela Git naming conventions like this **#162813038 Build API endpoint for fetching a question**. Follow this guide [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).

File: app/api/v1/questions/questions.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import flask
+from flask import request,jsonify,abort,make_response
+
+app = flask.Flask('__name__')
+app.config['DEBUG'] = True
+
+#add dummy questions
+questions = [{'id':1,
```
Comment (khwilo) [inline app/api/v1/questions/questions.py:8]: I would you like for you to consider placing the questions list into its equivalent question model class. Like this:
```python
QUESTIONS = []

class QuestionModel:
    def __init__(self, title, description):
        self.id = len(QUESTION)  + 1
        self.title = title
        self.description = description

    '''
    Add other methods to access the QUESTIONS list 
    like get_questions(), get_question_by_id(), etc.
    Depending on your implementation details, they methods may be
    class methods, instance methods or static methods.
    '''
```
**NOTE**:  _You need also to provide a method for converting the QuestionModel class to a dictionary._

File: app/api/v1/questions/questions.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            'What is computer science ?':'posted by Charles'},
+            {'id':2,
+            'Explain the importance of the Big O notation in computer science':'posted ny Ann'},
+            {'id':3,
+            'Define the term Algorithm':'posted by Goodwill'}]
+
+@app.route('/app/api/v1/questions/<int:question_id>',methods=['GET'])
+def showUser(question_id):
```
Comment (khwilo) [inline app/api/v1/questions/questions.py:16]: Consider placing the methods that deal with the **routes** in their own view file.

**Review by khwilo — COMMENTED** [review]
I have made a review of how you should structure your application. 
Use this application structure: 
screenshot from 2018-11-25 10-06-57
Use this [tutorial](https://scotch.io/tutorials/build-a-restful-api-with-flask-the-tdd-way) as a benchmark but consider using data structures instead of databases.

File: app/api/v1/questions/questions.py (hunk 1 above)
Comment (Barasa2018) [reply app/api/v1/questions/questions.py:8]: Thanks Khwilo for your feedback . i havent started on implimentation but i will take that point into consideration when i start. i was only checking if the endpoints are working

**Pull request merged**